app/models/grda_warehouse/data_source.rb

```diff
@@ -125,8 +126,10 @@ def self.authoritative_types
 
   private_class_method def self.has_access_to_data_source_through_viewable_entities(user, q, qc)
     data_source_table = quoted_table_name
-    viewability_table = GrdaWarehouse::UserViewableEntity.quoted_table_name
-    viewability_deleted_column_name = GrdaWarehouse::UserViewableEntity.paranoia_column
+    viewability_table = GrdaWarehouse::GroupViewableEntity.quoted_table_name
+    viewability_deleted_column_name = GrdaWarehouse::GroupViewableEntity.paranoia_column
+    group_ids = user.access_groups.pluck(:id)
+    group_id_query = "AND #{viewability_table}.#{qc.('access_group_id')} IN (#{group_ids.join(', ')})" unless group_ids.empty?
```

Comment:
What happens if group IDS is empty?  It looks like the SQL just ends up without the line, which would return all.  Is that correct?